File: Source/Obsequy.Model/Form/ConsumerRegistrationForm.cs
using FluentValidation.Results;
using Obsequy.Utility;

namespace Obsequy.Model
{
	public class ConsumerRegistrationForm
	{
		#region Properties

		public ConsumerMember Member { get; set; }
		public string Password { get; set; }
		public string ConfirmPassword { get; set; }
		public bool? HasAcceptedEULA { get; set; }

		#endregion

		public ConsumerRegistrationForm()
		{
		}

		public ValidationResult Validate(AccountSession accountSession, ValidationMode validationMode)
		{
			return ((new ConsumerRegistrationFormValidator(accountSession, validationMode)).Validate(this) as ValidationResult);
		}
	}
}

File: Source/Obsequy.Web/App/Forms/consumerRegistrationForm.js


angular.module('consumerRegistration-form', [])
	.factory('ConsumerRegistrationForm', ['ConsumerMember', function (ConsumerMember) {
		var ConsumerRegistrationForm = function (properties) {

 			angular.extend(this, {

 				member: new ConsumerMember(),
 				password: undefined,
 				confirmPassword: undefined,
 				hasAcceptedEULA: undefined,

 				update: function (properties) {
 					properties = properties || {};

					// update properties
 					this.member.update(properties.member);
 					this.password = properties.password;
 					this.confirmPassword = properties.confirmPassword;
 					this.hasAcceptedEULA = properties.hasAcceptedEULA;
				},

				toJSON: function () {
					return {
						member: this.member.toJSON(),
						password: this.password,
						confirmPassword: this.confirmPassword,
						hasAcceptedEULA: this.hasAcceptedEULA
					};
				}
			});

			this.update(properties);
		};

		return ConsumerRegistrationForm;
	}]);